packages/runtime/src/App.ce.vue

```diff
@@ -21,7 +21,20 @@ const shouldRender = computed(() => {
   return checkerResults.value.some((p) => p.diagnostics.length > 0)
 })
 
-const collapsed = ref<boolean>(!(props?.overlayConfig?.initialIsOpen ?? true))
+// collapsed state if requested (otherwise initially open)
+const collapsed = ref<boolean>(
+  props?.overlayConfig?.initialIsOpen === false || props?.overlayConfig?.initialIsOpen === 'error'
+)
+
+// if initialIsOpen is 'error': watch for results and open if there are errors
+if (props?.overlayConfig?.initialIsOpen === 'error') {
+  // checker results is initially empty, so need to watch for changes
+  watch(checkerResults, () => {
+    if (checkerResults.value.some((p) => p.diagnostics.some((d: any) => d.level === 1))) {
```

Comment:
The opposite actually. Level 1 designates error. Warning is level 0.

See `Badge.ce.vue` for example.

https://github.com/fi3ework/vite-plugin-checker/blob/vite-plugin-checker%400.6.1/packages/runtime/src/components/Badge.ce.vue#L23-L24


So this change will leave the overlay closed when the output is all warnings, but open it when there's at least one error. (set collapsed = false)

Reply:
Ah, I misunderstand that. Thanks for the correction. The issue is when `initialIsOpen` is set to `error`. The overlay will be uncollapsed when emitting a new error even if the user has manually collapsed it as `checkerResults` has been watched and will be run every time when changing. We need to make the `initialIsOpen` to be applied only once on the very beginning.

Reply:
@fi3ework good eye. I did grapple with that myself a good bit yesterday, and you're 100% correct. I thought this was the best I could do, but I think I've got an update to satisfy your concern. I'm also adding another playground test case.

Disclaimer: I'm not a Vue dev so I'm learning about things like `watch` and `ref` as I go.

What you'll see with my next change is that I'll keep the watcher around until the `checkerResults` length is non-zero. I initially thought that would be bad, because the code errors can be introduced later, but I actually think this is ok, because the overlay will only appear open at the same time that the badge first appears. So worst case here is the user's app loads clean with no badge visible, then they add bad code and the overlay appears in the open state at the same time that the badge would first appear. Because I unwatch, this only happens the first time.

Reply:
Interestingly, checker results are always empty initially. I've confirmed that much. `eslint` etc actually just takes time to run and report, and that's long enough for Vue to initially see that the results are empty. Errors and warnings that are already present on boot always come through the websocket with a delay.

Reply:
I think since there could be multiple checkers and the checker result are all async. We could using to collapsed state to control the panel. One is initial which will be configured by the config. The other one is user's collapsed state which is `null` or `undefined` at the beginning. And the final collapsed state will be defined as `userCollapsed ?? initialCollapsed`. By this, we could correctly set the initial collapsed state for async result, and supersede it by user's action.

Reply:
Good idea. 

I'm unavailable for a week, but I'll implement that after. 👍

Reply:
@fi3ework updated!